include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2263 @@
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuDownloadFromTexture(
+    SDL_GpuCopyPass *copyPass,
+    SDL_GpuTextureRegion *source,
+    SDL_GpuTextureTransferInfo *destination);
+
+/**
+ * Copies data from a buffer to a transfer buffer on the GPU timeline.
+ * This data is not guaranteed to be copied until the command buffer fence is signaled.
+ *
+ * \param copyPass a copy pass handle
+ * \param source the source buffer with offset and size
+ * \param destination the destination transfer buffer with offset
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuDownloadFromBuffer(
+    SDL_GpuCopyPass *copyPass,
+    SDL_GpuBufferRegion *source,
+    SDL_GpuTransferBufferLocation *destination);
+
+/**
+ * Ends the current copy pass.
+ *
+ * \param copyPass a copy pass handle
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuEndCopyPass(
+    SDL_GpuCopyPass *copyPass);
+
+/**
+ * Blits from a source texture region to a destination texture region.
+ * This function must not be called inside of any render, compute, or copy pass.
+ *
+ * \param commandBuffer a command buffer
+ * \param source the texture region to copy from
+ * \param destination the texture region to copy to
+ * \param filterMode the filter mode that will be used when blitting
+ * \param cycle if SDL_TRUE, cycles the destination texture if the destination texture slice is bound, otherwise overwrites the data.
+ *
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuBlit(
+    SDL_GpuCommandBuffer *commandBuffer,
+    SDL_GpuTextureRegion *source,
+    SDL_GpuTextureRegion *destination,
+    SDL_GpuFilter filterMode,
+    SDL_bool cycle);
+
+/* Submission/Presentation */
+
+/**
+ * Obtains whether or not a swapchain composition is supported by the GPU backend.
+ *
+ * \param device a GPU context
+ * \param window an SDL_Window
+ * \param swapchainComposition the swapchain composition to check
+ *
+ * \returns SDL_TRUE if supported, SDL_FALSE if unsupported (or on error)
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC SDL_bool SDLCALL SDL_GpuSupportsSwapchainComposition(
+    SDL_GpuDevice *device,
+    SDL_Window *window,
+    SDL_GpuSwapchainComposition swapchainComposition);
+
+/**
+ * Obtains whether or not a presentation mode is supported by the GPU backend.
+ *
+ * \param device a GPU context
+ * \param window an SDL_Window
+ * \param presentMode the presentation mode to check
+ *
+ * \returns SDL_TRUE if supported, SDL_FALSE if unsupported (or on error)
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC SDL_bool SDLCALL SDL_GpuSupportsPresentMode(
+    SDL_GpuDevice *device,
+    SDL_Window *window,
+    SDL_GpuPresentMode presentMode);
+
+/**
+ * Claims a window, creating a swapchain structure for it.
```

Comment:
I guess it's claiming a swapchain... or creating a swapchain for a window, and I think it's only 1 swapchain per window... it would be kind of nice to figure out where to put "swapchain" in there.